modules/javafx.graphics/src/main/java/com/sun/javafx/font/directwrite/DWFactory.java

```diff
@@ -151,4 +153,45 @@ static synchronized ID2D1Factory getD2DFactory() {
         }
         return D2D_FACTORY;
     }
+
+    private static native String regReadFontLink(String searchfont);
+    private static native String getEUDCFontFile();
+
+    /*
+     * Ignoring the primary on Windows - this should change some day.
+     */
+    public FontFallbackInfo getFallbacks(FontResource primaryResource) {
+        FontFallbackInfo info = new FontFallbackInfo();
+
+        String fontRegBuf = regReadFontLink("Tahoma");
+        if (fontRegBuf != null && fontRegBuf.length() > 0) {
+            // split registry data into null terminated strings
+            String[] fontRegList = fontRegBuf.split("\u0000");
+            int linkListLen = fontRegList.length;
+            for (int i=0; i < linkListLen; i++) {
+                String[] splitFontData = fontRegList[i].split(",");
+                int len = splitFontData.length;
+                String file = getPathNameWindows(splitFontData[0]);
+                String name = (len > 1) ? splitFontData[1] : null;
+                if (name != null && info.containsName(name)) {
+                    continue;
+                } else if (name == null && info.containsFile(file)) {
+                    continue;
+                }
+                info.add(name, file, null);
+            }
+        }
+
+        String eudcFontFile = getEUDCFontFile();
+        if (eudcFontFile != null) {
+            info.add(null, eudcFontFile, null);
+        }
+
+        // CJK Ext B Supplementary character fallbacks.
+        info.add("MingLiU-ExtB", getPathNameWindows("mingliub.ttc"), null);
+        info.add("Segoe UI Symbol", getPathNameWindows("seguisym.ttf"), null);
```

Comment:
a question: what if these fonts are *not* available in a particular system?  will the code break or it will be able to handle non-existing entry?

Reply:
No harm. Its quite common on initial windows install to not have these fonts.